lib/cylc/external_trigger.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2015 NIWA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import time
+import logging
+import Queue
+import Pyro.core
+import Pyro.errors
+import cylc.flags
+import cylc.cylc_pyro_client
+from cylc.task_id import TaskID
+from cylc.passphrase import passphrase
+
+
+PYRO_TARGET_NAME = 'external_event_broker'
+
+
+class PyroClient(object):
+    """Network client for sending external triggers to a suite."""
+
+    # TODO - configurable network retry parameters?
```

Comment:
Sounds like a sensible idea - depending on the connection etc for these things longer timeouts/variable try numbers may be appropriate.